### ISSUE TITLE
False positive: 'color-contrast' rule fails on intentionally invisible text

### ISSUE THREAD

**Issue by kylemh** [dequelabs/axe-core#2171]
**Expectation:** [CSS-Hidden Screen Reader Only text](https://accessibility.18f.gov/hidden-content/) or text with opacity intentionally set to zero with the express purpose of becoming visible on animation later should not trigger color contrast ratio violations.

**Actual:** Both above cases cause rule breaks

**Motivation:** Screen Reader Only text is my biggest motivation. Having been at multiple places where design gets final word on things, I want to provide queues and content to people hard-of-sight or devoid-of-sight. At these same places, I may be also dinged via automated tooling whose quota I may be mandated to meet.

```
axe-core version: 3.5.1
```
[Lighthouse uses 3.5.1, that is.](https://github.com/GoogleChrome/lighthouse/blob/master/package.json#L145)

---

_After template_

I saw this issue: https://github.com/dequelabs/axe-core/issues/1456

Which seems like my [React component's styling](https://github.com/OperationCode/front-end/blob/master/components/ScreenReaderOnly/ScreenReaderOnly.module.css) should behave the same (since width and height are at 1px)... but you will see many accessibility violations on this page: https://www.operationcode.org/code_schools

**Comment by straker** [comment]
The color contrast rule should filter out nodes that are hidden by CSS. However, looking at the page I see each card has multiple very light grey text nodes (which are usually proceeded by screen reader only text). These are the majority of failures from what I randomly selected.

![image](https://user-images.githubusercontent.com/2433219/80414715-af89d200-888e-11ea-9c0b-9167bf9134e6.png)

**Comment by kylemh** [comment]
Awkward. I presumed it was referencing a different node. Sorry that I had you do an audit for me. I'll fix that problem.
